chapter_introduction/index_vn.md

```diff
@@ -42,7 +52,19 @@ And when you are able to devise solutions that work $100\%$ of the time,
 *you should not be using machine learning*.
 -->
 
-*dịch đoạn phía trên*
+Để xây dựng được *bộ não* của ứng dụng này, ta phải bước qua tất cả mọi trường hợp ngách 
+mà chúng ta lường trước sẽ gặp phải, qua đó đặt ra những quy tắc thích hợp.
+Mỗi lần người dùng nhấn để thêm một món đồ vào giỏ hàng,
+ta thêm một trường vào bảng giỏ hàng trong cơ sở dữ liệu,
+liên kết ID của người dùng với ID của món hàng được yêu cầu.
+Mặc dù hầu như rất ít lập trình viên có thể làm đúng hết trong lần đầu tiên,
+(sẽ cần vài lần chạy kiểm tra để xử lý hết được những chỗ hiểm hóc),
+hầu như phần lớn là ta có thể lập trình được từ những nguyên tắc cơ bản nhất
```

Comment:
Cụm "first principle" chắc có nghĩa đặc biệt hơn là "những nguyên tắc cơ bản nhất". Mình nghĩ nó có gì đó liên quan đến "cổ điển" nhưng không chắc. Thử đọc thêm [wiki](https://en.wikipedia.org/wiki/First_principle) xem.
```suggestion
hầu như phần lớn ta có thể lập trình được từ những nguyên tắc cơ bản nhất
```

Reply:
Từ này được dùng ở nhiều ngữ cảnh, trong Toán học thì nó tương đương với Axiom/postulate nên mình đề xuất thành: "Định đề căn bản".

Reply:
Từ này chắc mình cần thêm thảo luận từ mọi người.
cc @ncthuong.

Reply:
Mình nghĩ "first principles" có nghĩa gần với "tiên đề" hoặc "định đề cơ bản" như @lkhphuc đề cập.

Tham khảo https://jamesclear.com/first-principles

> A first principle is a basic assumption that cannot be deduced any further. Over two thousand years ago, Aristotle defined a first principle as “the first basis from which a thing is known.”

Reply:
mình sẽ sửa tất cả "first principle" thành "định đề cơ bản" và thêm vào glossary.